Documentation/ApiOverview/FormProtection/Index.rst

```diff
@@ -1,150 +1,167 @@
+insert it as a hidden form element. The name of the form element does not
+matter; you only need it to get the form token for verifying it.
 
-.. code-block:: php
+..  code-block:: php
 
-   $formToken = TYPO3\CMS\Core\FormProtection\FormProtectionFactory::get()
-      ->generateToken('BE user setup', 'edit')
-   );
-   $this->content .= '<input type="hidden" name="formToken" value="' . $formToken . '" />';
+    $formToken = \TYPO3\CMS\Core\FormProtection\FormProtectionFactory::get()
+        ->generateToken('BE user setup', 'edit')
+    );
+    $this->content .= '<input type="hidden" name="formToken" value="' . $formToken . '">';
 
-The three parameters :code:`$formName`, :code:`$action` and :code:`$formInstanceName` can be arbitrary strings,
-but they should make the form token as specific as possible. For different forms
-(e.g. BE user setup and editing a tt_content record) or different records (with different UIDs)
-from the same table, those values should be different.
+The three parameters :php:`$formName`, :php:`$action` (optional) and
+:php:`$formInstanceName` (optional) can be arbitrary strings, but they should
+make the form token as specific as possible. For different forms (for example,
+BE user setup and editing a :sql:`tt_content` record) or different records (with
+different UIDs) from the same table, those values should be different.
 
-For editing a tt_content record, the call could look like this:
+For editing a :sql:`tt_content` record, the call could look like this:
 
-.. code-block:: php
+..  code-block:: php
 
-   $formToken = TYPO3\CMS\Core\FormProtection\FormProtectionFactory::get()->generateToken('tt_content', 'edit', $uid);
+    $formToken = \TYPO3\CMS\Core\FormProtection\FormProtectionFactory::get()
+        ->generateToken('tt_content', 'edit', $uid);
 
-At the end of the form, you need to persist the tokens. This makes sure that generated tokens get saved,
-and also that removed tokens stay removed:
+Finally, you need to persist the tokens. This makes sure that
+generated tokens get saved, and also that removed tokens stay removed:
 
-.. code-block:: php
+..  code-block:: php
 
-   TYPO3\CMS\Core\FormProtection\FormProtectionFactory::get()->persistTokens();
+    \TYPO3\CMS\Core\FormProtection\FormProtectionFactory::get()
+        ->persistTokens();
 
-In BE lists, it might be necessary to generate hundreds of tokens.
-So the tokens do not get automatically persisted after creation for performance reasons.
+In backend lists, it might be necessary to generate hundreds of tokens.
+So, the tokens are not automatically persisted after creation for performance
+reasons.
 
 When processing the data that has been submitted by the form,
 you can check that the form token is valid like this:
 
-.. code-block:: php
+..  code-block:: php
+
+    if ($dataHasBeenSubmitted &&
+        \TYPO3\CMS\Core\FormProtection\FormProtectionFactory::get()->validateToken(
```

Comment:
by using use statements the text would be much easier readable

Reply:
Agreed.